File: simulation/pressureSource.py
class PressureSource:
    # Object has default 0, 100, 0
    def __init__(self, initial_pressure=20, min_pressure=0, max_pressure=100):
        self.pressure = initial_pressure
        self.min_pressure = min_pressure
        self.max_pressure = max_pressure

    def increase_pressure(self, amount):
        self.pressure = min(self.pressure + amount, self.max_pressure)
        print(f"Pressure increased to {self.pressure}")

    def decrease_pressure(self, amount):
        self.pressure = max(self.pressure - amount, self.min_pressure)
        print(f"Pressure decreased to {self.pressure}")

    def set_pressure(self, value):
        if self.min_pressure <= value <= self.max_pressure:
            self.pressure = value
            print(f"Pressure set to {self.pressure}")
        else:
            print(f"Error: Pressure must be between {self.min_pressure} and {self.max_pressure}")

    def get_pressure(self):
        return self.pressure


# test code, must find a way to set source from javascript
if __name__ == "__main__":
    source = PressureSource(initial_pressure=50, max_pressure=200, min_pressure=0)

    source.increase_pressure(30) # increaase to 80
    source.decrease_pressure(50)  # decrease to 30
    source.set_pressure(150)  # set 150
    source.set_pressure(250)  # try to go above 150
    print(f"Current pressure: {source.get_pressure()}")

File: simulation/pressureSource.cpp
#include <iostream>
#include <algorithm>

class PressureSource{
    private:

    double currentPressure;
    double maxPressure;
    double minPressure;

    public:

    PressureSource(double initial, double max, double min): currentPressure(initial), maxPressure(max), minPressure(min) {} 

    void increasePressure(int amount){
        currentPressure = std::min(currentPressure + amount, maxPressure);
        std::cout << "Pressure increased to " << currentPressure << std::endl;;
    }

    void decreasePressure(int amount){
        currentPressure = std::max(currentPressure - amount, minPressure);
        std::cout << "Pressure decreased to " << currentPressure << std::endl;;
    }

    void setPressure(double value){
        if(minPressure <= value && maxPressure >= value){
            currentPressure = value;
            std::cout << "Pressure succesfully set to " << currentPressure << std::endl;;
    } else{
       std::cout <<"Error: Pressure must be between " << minPressure << " and " << maxPressure << std::endl;
        };
    }

    double getPressure() const{
        return currentPressure;
    }
};

/* testing numbers, remove comment bars to ensure it works

int main(){
    PressureSource test(50, 200, 0);
    test.increasePressure(30);
    test.decreasePressure(50);
    test.setPressure(150);
    test.setPressure(250);
    std::cout << "Current pressure: " << test.getPressure();

    return 0;
}
*/
